PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/classifiedexception/ClassifiedException.kt

```diff
@@ -0,0 +1,27 @@
+package com.microsoft.azure.hdinsight.common.classifiedexception
+
+import com.microsoft.azure.hdinsight.common.logger.ILogger
+import org.apache.commons.lang.exception.ExceptionUtils
+
+abstract class ClassifiedException : Throwable, ILogger {
```

Comment:
Just as discussed, use one parameter primary constructor is enough:
```suggestion
abstract class ClassifiedException(cause: Throwable) : Throwable, ILogger {
```

Reply:
I think we can use one parameter 'exp' to init 'cause' and 'message' property then in remoterunstate.kt ,we don't need to use exp.cause.message but exp.message.

---
In reply to: [226537464](https://github.com/Microsoft/azure-tools-for-java/pull/2100#discussion_r226537464) [](ancestors = 226537464)